packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3390,6 +3392,8 @@ export class ContainerRuntime
 			}),
 		};
 
+		this.channelCollection.notifyStagingMode(true);
+
 		return (this.stageControls = stageControls);
```

Comment:
Maybe set this.stageControls prior to notifying?  Not sure it specifically matters at the moment, but seems nice to complete internal state updates before calling out.

```suggestion
		this.stageControls = stageControls;
		this.channelCollection.notifyStagingMode(true);

		return this.stageControls;
```